### ISSUE TITLE
XCode7 (GM): compiler warnings

### ISSUE THREAD

**Issue by ogezue** [RestKit/RestKit#2301]
14 compiler warnings like "RKMIMETypeSerialization.m:41:17: Method override for the designated initializer of the superclass '-init' not found" found in RK 0.25

**Comment by segiddins** [comment]
A PR with a fix would be much appreciated!

**Comment by lucasecf** [comment]
Haven't been using RestKit with Swift, but it would be ok to remove the ```NS_DESIGNATED_INITIALIZER``` from the init declarations? Either this or implementing the designated init from the parent classes, calling the implemented designated inits, which does not make much sense

```objective-c
- (instancetype)init
{
    self = [self initWithMIMEType:nil serializationClass:nil];
    return self;
}
```

**Comment by segiddins** [comment]
No, the designated initializers should stay, and no args initializers need to be added that either throw an exception or call super.

**Comment by lucasecf** [comment]
The thing is, you really have to call the init marked as designated to get rid of the warning, so super is not enough. What you be a decent implementation for this inits with no args?

ex:
```objective-c
- (instancetype)init
{
    self = [self initWithMIMEType:nil serializationClass:nil];
    @throw [NSException exceptionWithName:NSInternalInconsistencyException
                                   reason:[NSString stringWithFormat:@"-init is not a valid initializer for the class %@", NSStringFromClass([self class])]
                                 userInfo:nil];
    return self;
}
```

**Comment by segiddins** [comment]
Yeah, except you'd want to throw the exception before delegating to the other initializer.

**Comment by lucasecf** [comment]
If you do that, you get a "Code will never be executed" warning in the line of the other initializer :/
